resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -393,7 +409,30 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
         val offerId = offer.getId.getValue
         val resources = remainingResources(offerId)
 
-        if (canLaunchTask(slaveId, resources)) {
+        var createTask = canLaunchTask(slaveId, resources)
+        if (hostToLocalTaskCount.nonEmpty) {
```

Comment:
I'm assuming that if `hostToLocalTaskCount` is not empty, there is locality info. I did another check for the executors to launch.

I did move the logic out to `satisfiesLocality()` and removed redundant executorLimit checks.